### ISSUE TITLE
Capturing calls to org.junit.jupiter.api.assertThrows<MyCustomException>

### ISSUE THREAD

**Issue by vincent-fuchs** [TNG/ArchUnit#1165]
hi,

in my Kotlin code base, I have quite a lot of calls to `org.junit.jupiter.api.assertThrows<MyCustomException>` , and depending on the package, I want to block it or not. 

I am a bit stuck on how to capture the exception type in an ArchUnit condition... 

so far, I am able to identify the places where assertThrows is called, but what other predicate can I use to filter on the exception type ?  

```kotlin
        val callOfAssertThrowsMyCustomException = callMethodWhere(
                target(name("assertThrows"))
         //then what ?
        )
```

is it even possible ? I have a doubt because when I try a different approach in debug mode, I am able to find the codeUnit I am interested in, but what I see is not reassuring : 

![image](https://github.com/TNG/ArchUnit/assets/9657583/4f8f6be9-e351-4fe1-9114-6df65c84d54f)

we see that the exception "real" class doesn't seem to be available, it's just a Class. 

If a more precise type information is not available to ArchUnit, it's going to be difficult, right ?

**Comment by hankem** [comment]
You're right; the generic type of [`org.junit.jupiter.api.Assertions.assertThrows`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertThrows(java.lang.Class,org.junit.jupiter.api.function.Executable)) does not show up in the byte code.
ArchUnit therefore cannot directly distinguish `assertThrows<MyCustomException>` from `assertThrows<AnyOtherException>`.

**Comment by hankem** [comment]
I wonder whether testing for [class objects referenced](https://www.javadoc.io/doc/com.tngtech.archunit/archunit/latest/com/tngtech/archunit/core/domain/JavaCodeUnit.html#getReferencedClassObjects()) from the same test method (it's not possible to attribute them directly to a method call) could be a useful workaround – even though this will have false positives and negatives.

This:
```java
ArchRule rule = noClasses().should().callMethodWhere(
        target(name("assertThrows")).and(describe("", call ->
                call.getOwner().getReferencedClassObjects().stream()
                        .map(ReferencedClassObject::getRawType)
                        .anyMatch(javaClass -> javaClass.isEquivalentTo(NullPointerException.class))
        ))
);
````
is able to distinguish those:
```java
@Test void test1() {
    assertThrows(NullPointerException.class, () -> requireNonNull(null));
}

@Test void test2() {
    assertThrows(IllegalArgumentException.class, () -> checkArgument(false));
}
```
